### ISSUE TITLE
Задание 4

### ISSUE THREAD

**Issue by nikitanut** [chesdenis/net-courses#13]
Не удаётся создать partial class. Насколько я понял, они должны находиться в одном namespace. Попробовал использовать тот же namespace, что и в .tt-файле (DataFile) - выводит предупреждение при создании объекта:

_Warning	CS0436	The type 'Artist' in 'C:\net-courses\Task4 - Partials\Nikita_Nuttunen\T4Example\T4ExampleConsoleApp\Program.cs' conflicts with the imported type 'Artist' in 'T4ExampleCore, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'C:\net-courses\Task4 - Partials\Nikita_Nuttunen\T4Example\T4ExampleConsoleApp\Program.cs'._	

Если создавать partial class Artist, то свойства класса из генерируемого файла недоступны. Доступны лишь созданные поля:

![2018-04-01_15-26-37](https://user-images.githubusercontent.com/30297448/38173193-7ef39d18-35c2-11e8-95d1-9946fe0ed9bb.png)

Если не создавать partial class Artist, то свойства становятся доступными.
Подскажите, пожалуйста, как в данном случае правильно использовать сгенерированный файл.

**Comment by BogdanVU** [comment]
Сделай using DataFile;

**Comment by BogdanVU** [comment]
The same problem)

**Comment by BogdanVU** [comment]
Возможно ли использование partial класса, если он в разных проектах одного решения?

**Comment by Chesanita** [comment]
All partial-type definitions meant to be parts of the same type must be defined in the same assembly and the same module (.exe or .dll file). Partial definitions cannot span multiple modules.
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods

Это из официальной документации. Так что если я правильно понял - то partial классы должны быть объявлены в одном решении, одной сборке. И не получится часть написать в одной сборке, а часть подключить из библиотеки.

**Comment by chesdenis** [comment]
Да, скорее всего мы ограничены в пределах одной библиотеки. Можно было попытаться добавить ссылку на проект как SharedProject, но шаблоны tt в ней не работают. Потому единственный вариант это объявлять partial в этой же сборке.  
![image](https://user-images.githubusercontent.com/15965781/38191561-ea4d2b6a-3671-11e8-9739-c5baabd2a775.png)
